core/src/test/scala/unit/kafka/server/FetchRequestTest.scala

```diff
@@ -285,8 +315,8 @@ class FetchRequestTest extends BaseFetchRequestTest {
 
     // Check follower error codes
     val followerId = TestUtils.findFollowerId(topicPartition, servers)
-    assertResponseErrorForEpoch(Errors.NONE, followerId, Optional.empty())
-    assertResponseErrorForEpoch(Errors.NONE, followerId, Optional.of(secondLeaderEpoch))
+    assertResponseErrorForEpoch(Errors.NOT_LEADER_OR_FOLLOWER, followerId, Optional.empty())
+    assertResponseErrorForEpoch(Errors.NOT_LEADER_OR_FOLLOWER, followerId, Optional.of(secondLeaderEpoch))
```

Comment:
Note: The epoch is correctly validated here but the request is now rejected because the cluster does not have a replica selector.

Reply:
I'm afraid this will cause some confusion or possibly hide some potential bug.
I'm thinking we can override the beforeEach to create a test broker with replica selector config for this test to avoid `NOT_LEADER_OR_FOLLOWER` error. ex:

```java
class FetchRequestTest extends BaseFetchRequestTest {
  var shouldCreateReplicaSelector = false

  @BeforeEach
  override def setUp(testInfo: TestInfo): Unit = {
    if (testInfo.getDisplayName().equals("testCurrentEpochValidation")) {
      shouldCreateReplicaSelector = true
    }
    super.setUp(testInfo)
  }

  override def generateConfigs: Seq[KafkaConfig] = {
    TestUtils.createBrokerConfigs(1, zkConnect).map( p => {
      if (shouldCreateReplicaSelector)
        p.setProperty("replica.selector.class", "mock.class.name")
      KafkaConfig.fromProps(p)
    })
  }
```

WDYT?

Reply:
Yeah, I was thinking about this as well. The point is that this is the correct and expected behavior in this case so I am inclined to keep it as it is now. However, it would be great to have a test suite with a replica selector enabled. I can do this as a follow-up.

Reply:
Sounds good to me! Thanks.